hi-ml/src/health_ml/deep_learning_config.py

```diff
@@ -0,0 +1,451 @@
+            raise ValueError(f"Cross validation split index must be -1 for a non cross validation run, "
+                             f"found num_crossval_splits = {self.num_crossval_splits} "
+                             f"and crossval_split_index={self.crossval_split_index}")
+
+    @property
+    def is_running_in_aml(self) -> bool:
+        """
+        Whether the current run is executing inside Azure ML
+
+        :return: True if the run is executing inside Azure ML, or False if outside AzureML.
+        """
+        return is_running_in_azure_ml(RUN_CONTEXT)
+
+    @property
+    def perform_cross_validation(self) -> bool:
+        """
+        True if cross validation will be be performed as part of the training procedure.
+
+        :return:
+        """
+        return self.num_crossval_splits > 1
+
+    def get_effective_random_seed(self) -> int:
+        """
+        Returns the random seed set as part of this configuration. If the configuration corresponds
+        to a cross validation split, then the cross validation fold index will be added to the
+        set random seed in order to return the effective random seed.
+
+        :return:
+        """
+        seed = self.random_seed
+        if self.perform_cross_validation:
+            # offset the random seed based on the cross validation split index so each
+            # fold has a different initial random state.
+            seed += self.crossval_split_index
+        return seed
+
+
+class DatasetParams(param.Parameterized):
+    azure_datasets: List[str] = param.List(default=[], class_=str,
+                                           doc="If provided, the ID of one or more datasets to use when running in"
+                                               " AzureML.This dataset must exist as a folder of the same name in the"
+                                               " 'datasets' container in the datasets storage account. This dataset"
+                                               " will be mounted and made available at the 'local_dataset' path"
+                                               " when running in AzureML.")
+    local_datasets: List[str] = param.List(default=[], class_=Path,
+                                           doc="A list of one or more paths to the dataset to use, when training"
+                                               " outside of Azure ML.")
+    dataset_mountpoints: List[str] = param.List(default=[], class_=Path,
+                                                doc="The path at which the AzureML dataset should be made available "
+                                                    "via mounting or downloading. This only affects jobs running in "
+                                                    "AzureML. If empty, use a random mount/download point.")
+
+    def validate(self) -> None:
+        if not self.azure_datasets and self.local_datasets is None:
```

Comment:
local_datasets cannot be None as per your type annotations. `if (not azure_datasets) and (not local_datasets)`